tests/unit/lib/charms/kubernetes_charm_libraries/v0/test_multus.py

```diff
@@ -0,0 +1,579 @@
+            name=statefulset_name,
+            network_annotations=network_annotations,
+            containers_requiring_net_admin_capability=[],
+        )
+
+        assert is_patched
+
+    @patch("lightkube.core.client.Client.get")
+    def test_given_annotations_are_already_present_and_security_context_is_missing_when_statefulset_is_patched_then_returns_false(  # noqa: E501
+        self, patch_get
+    ):
+        container_name = "whatever container"
+        statefulset_name = "whatever name"
+        network_annotations = [
+            NetworkAnnotation(interface="whatever interface 1", name="whatever name 1"),
+            NetworkAnnotation(interface="whatever interface 2", name="whatever name 2"),
+        ]
+        patch_get.return_value = StatefulSet(
+            spec=StatefulSetSpec(
+                selector=LabelSelector(),
+                serviceName="",
+                template=PodTemplateSpec(
+                    metadata=ObjectMeta(
+                        annotations={
+                            "k8s.v1.cni.cncf.io/networks": json.dumps(
+                                [
+                                    network_annotation.dict()
+                                    for network_annotation in network_annotations
+                                ]
+                            )
+                        },
+                    ),
+                    spec=PodSpec(
+                        containers=[
+                            Container(
+                                name=container_name,
+                                securityContext=SecurityContext(
+                                    capabilities=Capabilities(add=[], drop=[])
+                                ),
+                            )
+                        ]
+                    ),
+                ),
+            )
+        )
+
+        is_patched = self.kubernetes_multus.statefulset_is_patched(
+            name=statefulset_name,
+            network_annotations=network_annotations,
+            containers_requiring_net_admin_capability=[container_name],
+        )
+
+        assert not is_patched
+
+    @patch("lightkube.core.client.Client.delete")
+    def test_given_when_delete_nad_then_k8s_delete_is_called(self, patch_delete):
+        nad_name = "whatever name"
+
+        self.kubernetes_multus.delete_network_attachment_definition(name=nad_name)
+
+        patch_delete.assert_called_with(
+            res=NetworkAttachmentDefinition, name=nad_name, namespace=self.namespace
+        )
+
+
+class _TestCharmNoNAD(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.kubernetes_multus = KubernetesMultusCharmLib(
+            charm=self,
+            network_attachment_definitions=[],
+            network_annotations_func=self._network_annotations_func,
+            containers_requiring_net_admin_capability=[],
+        )
+
+    def _network_annotations_func(self) -> list[NetworkAnnotation]:
```

Comment:
I'm having this error message from Pylance `Subscript for class "list" will generate runtime exception; enclose type annotation in quotesPylance` I think we should use `List` instead for proper type checking.
```suggestion
    def _network_annotations_func(self) -> List[NetworkAnnotation]:
```

Reply:
This came from a @ghislainbourgeois comment. Ultimately, with Python versions > 3.9, using `list` is fine. Reference [here](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)